processors/src/main/resources/DeltaBuilderBody.vm

```diff
@@ -61,23 +61,27 @@
     #end
     public ${return_type} set$alias(final $field.type newValue) {
         // first check if we already have a delta of this field
-        ${field.boxedType} curValue;
 
         if(deltaMap().containsKey("${field.name}")) {
-            curValue = ($field.boxedType) deltaMap().get("${field.name}").getOldValue();
+            ${field.boxedType} curValue = ($field.boxedType) deltaMap().get("${field.name}").getOldValue();
+            Delta.OP op = compare(curValue, newValue);
+            if(op != Delta.OP.NOOP) {
+                addDelta(new Delta(op, "${field.name}", curValue, newValue));
+            } else {
+                removeDelta("${field.name}");
```

Comment:
Not only that, I want to do the removeDelta only when the deltaMap() previously contained that key.  Otherwise there is no need to call removeDelta (which is an extra cost if not needed).